src/include/migraphx/op/where.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_WHERE_HPP
+#define MIGRAPHX_GUARD_OPERATORS_WHERE_HPP
+
+#include <array>
+#include <migraphx/argument.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/streamutils.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/op/normalize_attribute.hpp>
+#include <cmath>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct where
+{
+    std::string name() const { return "where"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.same_dims();
+        return inputs[1];
```

Comment:
This needs to use the same logic in the binary.hpp for inputs[1] and inputs[2].

Reply:
The specification is that cond, input[1] and input[2] can all be broadcasted, so this one is more complex than the binary operators since this have three inputs.

Reply:
In general, we dont need to consider the permutation of the condition since that will never be whats returned from `where`.